pandas-stubs/_typing.pyi

```diff
@@ -392,74 +479,40 @@ AstypeArg: TypeAlias = (
     | DtypeObj
 )
 
-# filenames and file-like-objects
-AnyStr_cov = TypeVar("AnyStr_cov", str, bytes, covariant=True)
-AnyStr_con = TypeVar("AnyStr_con", str, bytes, contravariant=True)
-
-class BaseBuffer(Protocol):
-    @property
-    def mode(self) -> str: ...
-    def seek(self, offset: int, whence: int = ..., /) -> int: ...
-    def seekable(self) -> bool: ...
-    def tell(self) -> int: ...
-
-class ReadBuffer(BaseBuffer, Protocol[AnyStr_cov]):
-    def read(self, n: int = ..., /) -> AnyStr_cov: ...
-
-class WriteBuffer(BaseBuffer, Protocol[AnyStr_con]):
-    def write(self, b: AnyStr_con, /) -> Any: ...
-    def flush(self) -> Any: ...
-
-class ReadPickleBuffer(ReadBuffer[bytes], Protocol):
-    def readline(self) -> bytes: ...
-
-class ReadCsvBuffer(ReadBuffer[AnyStr_cov], Protocol[AnyStr_cov]):
-    def __iter__(self) -> Iterator[AnyStr_cov]: ...
-    def fileno(self) -> int: ...
-    def readline(self) -> AnyStr_cov: ...
-    @property
-    def closed(self) -> bool: ...
-
-class WriteExcelBuffer(WriteBuffer[bytes], Protocol):
-    def truncate(self, size: int | None = ..., /) -> int: ...
-
-FilePath: TypeAlias = str | PathLike[str]
+# DtypeArg specifies all allowable dtypes in a functions its dtype argument
+DtypeArg: TypeAlias = Dtype | Mapping[Hashable, Dtype]
+DtypeObj: TypeAlias = np.dtype[np.generic] | ExtensionDtype
 
-_T_co = TypeVar("_T_co", covariant=True)
+# converters
+ConvertersArg: TypeAlias = dict[Hashable, Callable[[Dtype], Dtype]]
 
-class SequenceNotStr(Protocol[_T_co]):
-    @overload
-    def __getitem__(self, index: SupportsIndex, /) -> _T_co: ...
-    @overload
-    def __getitem__(self, index: slice, /) -> Sequence[_T_co]: ...
-    def __contains__(self, value: object, /) -> bool: ...
-    def __len__(self) -> int: ...
-    def __iter__(self) -> Iterator[_T_co]: ...
-    def index(self, value: Any, start: int = ..., stop: int = ..., /) -> int: ...
-    def count(self, value: Any, /) -> int: ...
-    def __reversed__(self) -> Iterator[_T_co]: ...
+# parse_dates
+ParseDatesArg: TypeAlias = (
+    bool | list[Hashable] | list[list[Hashable]] | dict[Hashable, list[Hashable]]
+)
```

Comment:
Might need HashableT (and Mapping/dict + HashableT2 for the dict values)

Reply:
ok